app/scripts/controllers/locationPickerMap.ctrl.js

```diff
@@ -18,41 +19,48 @@ angular.module('yourStlCourts').controller('LocationPickerMapCtrl', function ($s
     dashArray: '',
     fillOpacity: 0.5
   };
+
   var outlineStyle = {
     weight: 3,
     color: 'blue',
     fillColor:'',
     dashArray: '',
     fillOpacity: 0
   };
+
   function highlightFeature(e) {
-    if (!ctrl.selectedMunicipalitiesObj.idInArray(e.target.feature.id)) {
+    var municipality = getMunicipalityFromMapName(e.target.feature.properties.municipality);
+    if(!isMunicipalitySelected(municipality.id)) {
       var layer = e.target;
       layer.setStyle(outlineStyle);
 
       if (!L.Browser.ie && !L.Browser.opera && !L.Browser.edge) {
         layer.bringToFront();
       }
     }
-    ctrl.mousedOverMunicipality = mapMuniObj(e.target.feature.id,e.target.feature.properties.municipality);
+    ctrl.mousedOverMunicipality = municipality;
   }
 
   function resetHighlight(e) {
-    if (!ctrl.selectedMunicipalitiesObj.idInArray(e.target.feature.id)) {
+    var municipality = getMunicipalityFromMapName(e.target.feature.properties.municipality);
+    if(!isMunicipalitySelected(municipality.id)) {
       geoJson.resetStyle(e.target);
     }
-    ctrl.mousedOverMunicipality = "";
+    ctrl.mousedOverMunicipality = null;
   }
 
   function selectMunicipality(e) {
-    if (ctrl.selectedMunicipalitiesObj.municipalityClicked(mapMuniObj(e.target.feature.id,e.target.feature.properties.municipality))){
+    var municipality = getMunicipalityFromMapName(e.target.feature.properties.municipality);
+    if(isMunicipalitySelected(municipality.id)) {
+      removeMuncipality(municipality);
+      geoJson.resetStyle(e.target);
+    } else {
+      addMunicipality(municipality);
```

Comment:
I'm concerned here that by returning municipality from getMunicipalityFromMapName a few items may be overlooked.  First you are checking if the municipality is selected by checking for municipality.id which I believe is now the DB ID field for the municipality and not the map's ID field for the municipality.  The concern is that "unincorporated" is not all mapped to one ID.  So if you click on more than one unincorporated area, you do want it to add the highlight to the map but you don't want it to add a second "St. Louis County" to the map (you only need one to appear).  Likewise, you don't want to unselect or remove "St. Louis County" if you click on a different unincorporated area of the map.  You only want it to disappear if all "unincorporated" areas are removed.